src/libraries/System.Private.CoreLib/src/System/IO/Strategies/UnixFileStreamStrategy.cs

```diff
@@ -15,5 +15,31 @@ internal UnixFileStreamStrategy(string path, FileMode mode, FileAccess access, F
             base(path, mode, access, share, options, preallocationSize, unixCreateMode)
         {
         }
+
+        internal override SafeFileHandle SafeFileHandle
+        {
+            get
+            {
+                if (CanSeek)
+                {
+                    // The base SafeFileHandle getter synchronizes the file offset before exposing the handle
+                    // since the in-memory position may be out-of-sync with the actual file position.
+                    // However, some files such as certain pseudofiles on AzureLinux may report CanSeek = true
+                    // but still fail with ESPIPE when attempting to seek. We tolerate this specific error
+                    // to match the behavior in RandomAccess where we ignore seek failures on certain file types.
```

Comment:
In RandomAccess we ignore pwrite and pread failures and fall back to write and read.

```suggestion
                    // to match the behavior in RandomAccess where we ignore pwrite and pread failures on certain file types.
```